pages/explore/components/ExploreItemCarousel.tsx

```diff
@@ -76,25 +79,47 @@ function ExploreItemCarousel({
               // })}
             >
               <View style={styles.descriptionContainer}>
-                <Text style={styles.title}> {item.title} </Text>
+                <View style={styles.titleContainer}>
+                  <Text
+                    style={styles.title}
+                    numberOfLines={2}
+                    ellipsizeMode="tail"
+                  >
+                    {" "}
+                    {item.title}{" "}
+                  </Text>
+                </View>
                 <Text style={styles.price}>
-                  {" "}
                   ${item.price}
                   <Text style={[styles.price, { fontSize: 14 }]}>
-                    /night
+                    {" "}
+                    / night
                   </Text>{" "}
                 </Text>
               </View>
             </Pressable>
-
-            <ImageCarousel
-              width={cardWidth}
-              height={cardHeight}
-              imagesToShow={item.images ?? []}
-              navigation={navigation}
-              item={item}
-              page={false}
-            />
+            <Pressable
+              key={index}
+              onPress={() =>
+                navigation.navigate("Detail", {
+                  item: item,
+                  navigation: navigation,
+                })
+              }
+            >
+              <ImageCarousel
+                width={cardWidth}
+                height={cardHeight}
+                imagesToShow={
+                  item.images ?? [
+                    "../../../assets/images/default-listing-image.png",
```

Comment:
This does not work because the ?? operator only applies when the LHS is undefined or null. But in this case, item.images will only be an empty array or a filled array. 

So the fix would be to upload the default-listing-image into our firebase images storage and add a reference to that object into the images array whenever the user does not include any images.